### DISCUSSION TITLE
query about passing parameters using the "InplaceAdd" instruction

### DISCUSSION THREAD

**Discussion by patriciomh1965** [serenity-is/Serenity#5703] [Help]
![Panta_ejemplo](https://user-images.githubusercontent.com/42218266/117582242-0c89ac00-b0cf-11eb-81d5-4f4f2448afc0.png)

![Panta_ejemplo 2](https://user-images.githubusercontent.com/42218266/117582533-69d22d00-b0d0-11eb-88aa-40910ae20c02.png)
Hello Community

 I must create a new record within a form using the "InplaceAdd" instruction, I must pass some parameters to it. Look in the "Issues" and I have not achieved the solution, below I attach an image and the instruction that I am using in XYZRow:

Next I show you the instruction in XYZRow ;

public sealed class TaCncPlanRow : Row, IIdRow, INameRow
......
......
  //
   [DisplayName("Origen"), Column("pln_origen"), ForeignKey("[dbo].[ta_cnc_param_origen]", "ori_numero"), LeftJoin("jOrigen"), TextualField("NombreOrigen"), DefaultValue(1)]
        [LookupEditor(typeof(TaCncParamOrigenFiltroRow))]
        public Int32? PlnOrigen
        {
            get { return Fields.PlnOrigen[this]; }
            set { Fields.PlnOrigen[this] = value; }
        }
        //
        [DisplayName("Origen"), Expression("jOrigen.[ori_descripcion]")]
        public String NombreOrigen
        {
            get { return Fields.NombreOrigen[this]; }
            set { Fields.NombreOrigen[this] = value; }
        }

        [DisplayName("Compañia"), Column("pln_cia"), ForeignKey(typeof(TaCncConfigCiaRow)), LeftJoin("cia")]
        [LookupEditor(typeof(TaCncConfigCiaRow))]
        public Int32? PlnCia
        {
            get { return Fields.PlnCia[this]; }
            set { Fields.PlnCia[this] = value; }
        }
        [DisplayName("Compañia")]
        [Origin("cia"), MinSelectLevel(SelectLevel.List)]
        public String CiaGlosa
        {
            get { return Fields.CiaGlosa[this]; }
            set { Fields.CiaGlosa[this] = value; }
        }
        //
        [DisplayName("Poliza"), Column("pln_poliza"), ForeignKey(typeof(TaCncConfigPolizaNuevaRow)), LeftJoin("pol"), LookupInclude, DefaultValue("")]
        [LookupEditor(typeof(TaCncConfigPolizaNuevaRow), CascadeFrom = "PlnCia", CascadeField = "PolCia", InplaceAdd = true)]      
        public Int32? PlnPoliza
        {
            get { return Fields.PlnPoliza[this]; }
            set { Fields.PlnPoliza[this] = value; }
        }
        [Origin("pol"), DisplayName("Poliza")]
        public String PolOrigenPoliza
        {
            get { return Fields.PolOrigenPoliza[this]; }
            set { Fields.PolOrigenPoliza[this] = value; }
        }
        //

  public sealed class TaCncConfigPolizaNuevaRow : Row, IIdRow, INameRow
......
.....

 [DisplayName("Origen"), Column("pol_origen"), DefaultValue(0), LookupInclude]
        [LookupEditor(typeof(TaCncParamOrigenRow))]
        public Int32? PolOrigen
        {
            get { return Fields.PolOrigen[this]; }
            set { Fields.PolOrigen[this] = value; }
        }

 [DisplayName("Compania"), Column("pol_compania"), DefaultValue(0), LookupInclude]
        [LookupEditor(typeof(TaCncConfigCiaRow))]
        public Int32? PolCia
        {
            get { return Fields.PolCia[this]; }
            set { Fields.PolCia[this] = value; }
        }
....
....

The parameters that I must pass are:
PlnOrigin and PlnCia

Already grateful for the help you can give me

Greetings.

**Comment by minhhungit** [comment]
check this issue https://github.com/serenity-is/Serenity/issues/4122#issuecomment-445362937

```javascript
this.form.SomeField.onInitNewEntity = entity => {
  entity.SomeInfoField = "someValue";
}
```

**Comment by patriciomh1965** [comment]
Hello

Thanks

To pass two or more parameters?
How should the institution be created

**Reply by minhhungit** [reply]
if you can pass 1 param then why you can not pass 2 params, why just keep asking without even do not try to apply my suggestion or spend a bit time to think ?

**Reply by minhhungit** [reply]
I don't want to be a rude guy but it's very easy, instead of passing 1 param like this:

```javascript
this.form.SomeField.onInitNewEntity = entity => {
  entity.SomeInfoField = "someValue";
}
```

then just add more lines for more params:

```javascript
this.form.SomeField.onInitNewEntity = entity => {
  entity.SomeInfoField = "value 1";
  entity.YourSecondParam = "value 2";
  entity.SomethingElseParam = "value 3";
}
```

It is not hard

**Comment by patriciomh1965** [comment]
Hi jin

I had done it that way, believe me I did. Sorry for the inconvenience.

Patrick
Example :

//this.form.PolNumero.onInitNewEntity = x => x.PolCia = this.form.PlnCia;
            //this.form.PolNumero.onInitNewEntity = x => x.PolOrigen = this.form.PlnOrigen;
            this.form.PprNumero.onInitNewEntity = entity => {
                entity.SomeInfoField = this.form.TprNumero;
            }
            // this.form.PolNumero.onInitNewEntity = x => x.PolOrigen = this.form.PlnOrigen; x =>  x.PolCia = this.form.PlnCia;         
            // this.form.PprNumero.onInitNewEntity = x => x.PprTipoProducto = this.form.TprNumero;
            // this.form.PprNumero.onInitNewEntity = x => x.PprTipoProducto = 163;

            // TRASPASA PARAMETROS A LA NUEVA ENTIDAD (INPLACED)
            this.form.PolNumero.onInitNewEntity = entity => {
                entity.SomeInfoField  = this.form.PlnCia,
                    entity.SomeInfoField  = this.form.PlnOrigen;
            }

            //this.form.PolNumero.inplaceCreateClick(e => {
            //    var dlg = new ServiciosProducto.TaCncConfigPolizaNuevaDialog();

            //    this.initDialog(dlg);

            //    dlg.loadEntityAndOpenDialog(<ServiciosProducto.TaCncConfigPolizaNuevaRow>{
            //        PolOrigen: this.form.PlnOrigen,
            //        PolCia: this.form.PlnCia
            //    });
            //});

**Reply by minhhungit** [reply]
Just put the code on your **Planes** dialog (ex in afterLoadEntity), and get those values on **Poliza** dialog `afterLoadEntity`
